implementations/code_df/issue.py

```diff
@@ -0,0 +1,55 @@
+from metric import Metric
+import utils
+
+
+class Issue(Metric):
+
+    def __init__(self, items, date_range=(None, None)):
+        """
+        Initilizes self.df, the dataframe with one commit per row.
+
+        :param items: A list of dictionaries.
+            Each item is a Perceval dictionary, obtained from a JSON
+            file or from Perceval directly.
+
+        :param date_range: A tuple which represents the period of interest
+            It is of the form (since, until), where since and until are
+            strings. Either, or both can be None. If, for example, since
+            is None, that would mean that all commits from the first commit
+            to the commit who last falls inside the until range will be
+            included.
+        """
+
+        self.since, self.until = date_range
+        super().__init__(items)
+
+    def _flatten(self, item):
+        """
+        Flatten a raw issue fetched by Perceval into a flat dictionary.
+
+        A list with a single flat directory will be returned.
+        That dictionary will have the elements we need for computing metrics.
+        The list may be empty, if for some reason the commit should not
+        be considered.
+
+        :param item: raw item fetched by Perceval (dictionary)
+        :returns:   list of a single flat dictionary
+        """
+
+        creation_date = utils.str_to_date(item['data']['created_at'])
+        if self.since and (self.since > creation_date):
+            return []
+
+        if self.until and (self.until < creation_date):
+            return []
+
+        flat = {
+            'repo': item['origin'],
+            'hash': item['data']['id'],
+            'category': "issue",
+            'author': item['data']['user']['login'],
+            'created_date': creation_date,
+            'current_status': item['data']['state']
+        }
+
+        return [flat]
```

Comment:
why do you need to return a list here?

Reply:
This is part of Jesus' design. If you have a look at a part of the metric (root) class: 
```python
    def __init__(self, items):

        flat_items = []
        for item in items:
            flat_items.extend(self._flatten(item))
        self.df = pd.DataFrame(flat_items)
```

the flat_items list is extended by each flattened item. Hence, `_flatten` must pass a list so the `extend` can work

Reply:
The idea is that in some cases, _flat could return more than one row. For example, if you need to "convert" a commit in information for each file touched by that commit.